src/pages/TopicArticles.jsx

```diff
@@ -0,0 +1,20 @@
+import TopicBar from "../Components/TopicBar";
+import ArticleContainer from "../Components/ArticleContainer";
+import {useParams} from "react-router-dom"
+
+const TopicArticles = ({articles}) => {
+    const {topic} = useParams()
+
+    let filteredArticles = articles.filter((article) => article.topic === `${topic}`)
```

Comment:
You have an endpoint for a topic query, which would serve up articles by topic.   I understand what you've done here and it's good logic.  But you could just make use of your existing articles component, and getArticles function, just passing in a topic as parameter!  I suggest giving that a go!